### PR TITLE
feat: Adds textarea component to use in forms

### PR DESCRIPTION
**Pull request by nithindavid**
**Description**
Adds form input component that needs to be used across onboarding wizard.

image

image

- [x] New feature (non-breaking change which adds functionality)
- [ ] This change requires a documentation update

## Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my code
- [x] My changes generate no new warnings

### CODE REVIEW

File: app/javascript/v3/components/Form/Textarea.vue [hunk 1]
```diff
@@ -0,0 +1,90 @@
...
+      :rows="rows"
+      :class="{
+        'focus:ring-red-600 ring-red-600': hasError,
+        'dark:ring-slate-600 dark:focus:ring-woot-500 ring-slate-200':
+          !hasError,
+        'px-3 py-3': spacing === 'base',
+        'px-3 py-2': spacing === 'compact',
+        'resize-none': !allowResize,
+      }"
+      class="block w-full border-none rounded-xl shadow-sm appearance-none outline outline-1 outline-slate-200 dark:outline-slate-800 focus:outline-none focus:outline-0 text-slate-900 dark:text-slate-100 placeholder:text-slate-400 focus:ring-2 focus:ring-woot-500 sm:text-sm sm:leading-6 dark:bg-slate-800"
```
Comment (iamsivin) [inline app/javascript/v3/components/Form/Textarea.vue:25]: @nithindavid As we discussed before for the Input component that the tailwind `ring` is not working on production, can we use same way that we used for input.

**Review by iamsivin — COMMENTED** [review]
@nithindavid LGTM, Added a minor comment, please take a look.

File: app/javascript/v3/components/Form/Input.vue [hunk 2]
```diff
@@ -28,7 +28,7 @@
         'px-3 py-2 mb-0': spacing === 'compact',
         'pl-9': icon,
       }"
-      class="block w-full border-none shadow-sm appearance-none rounded-xl outline outline-1 focus:outline-2 text-slate-900 dark:text-slate-100 placeholder:text-slate-400 sm:text-sm sm:leading-6 dark:bg-slate-800"
+      class="block w-full border-none shadow-sm appearance-none rounded-xl outline outline-1 focus:outline focus:outline-2 text-slate-900 dark:text-slate-100 placeholder:text-slate-400 sm:text-sm sm:leading-6 dark:bg-slate-800"
```
Comment (iamsivin) [inline app/javascript/v3/components/Form/Input.vue:31]: @nithindavid Do we need this change here?

File: app/javascript/v3/components/Form/Textarea.vue [hunk 3]
```diff
@@ -0,0 +1,90 @@
...
+      :rows="rows"
+      :class="{
+        'focus:outline-red-600 outline-red-600': hasError,
+        'dark:outline-slate-600 dark:focus:outline-woot-500 outline-slate-200 focus:outline-woot-500':
+          !hasError,
+        'px-3 py-3': spacing === 'base',
+        'px-3 py-2': spacing === 'compact',
+        'resize-none': !allowResize,
+      }"
+      class="block w-full border-none rounded-xl shadow-sm appearance-none outline outline-1 focus:outline focus:outline-2 text-slate-900 dark:text-slate-100 placeholder:text-slate-400 dark:bg-slate-800"
```
Comment (iamsivin) [inline app/javascript/v3/components/Form/Textarea.vue:25]: @nithindavid Do we need `focus:outline` here? Already added `focus:outline-2` and `outline`

**Review by iamsivin — COMMENTED** [review]
@nithindavid Please take a look?

File: app/javascript/v3/components/Form/Input.vue (hunk 2 above)
Comment (nithindavid) [reply app/javascript/v3/components/Form/Input.vue:31]: I'll make it a separate PR

File: app/javascript/v3/components/Form/Textarea.vue (hunk 3 above)
Comment (nithindavid) [reply app/javascript/v3/components/Form/Textarea.vue:25]: `outline` is for `outline-style: solid;` and `outline-2` is the width of the outline.

**Review by iamsivin — APPROVED** [review]

File: app/javascript/v3/components/Form/Textarea.vue [hunk 4]
```diff
@@ -0,0 +1,90 @@
...
+    },
+    dataTestid: {
+      type: String,
+      default: '',
+    },
+    spacing: {
+      type: String,
+      default: 'base',
+      validator: value => ['base', 'compact'].includes(value),
+    },
```
Comment (pranavrajs) [inline app/javascript/v3/components/Form/Textarea.vue:82]: This change appears to have been made without much consideration. It simply modifies the padding for the text area. Are there alternative options for the text area in the design?

File: app/javascript/v3/components/Form/Textarea.vue (hunk 3 above)
Comment (pranavrajs) [reply app/javascript/v3/components/Form/Textarea.vue:25]: Why do we change the width of the outline on focus>

Comment (pranavrajs) [inline app/javascript/v3/components/Form/Textarea.vue:25]: There is no background class for non-dark mode. It only has `dark:bg-slate-800`.

**Review by pranavrajs — CHANGES_REQUESTED** [review]
Minor comments, please take a look at comments above.

File: app/javascript/v3/components/Form/Textarea.vue (hunk 3 above)
Comment (nithindavid) [reply app/javascript/v3/components/Form/Textarea.vue:25]: This is to have a distinguished focus effect.

Comment (pranavrajs) [reply app/javascript/v3/components/Form/Textarea.vue]: Why is not ring working on production?

File: app/javascript/v3/components/Form/Textarea.vue [hunk 5]
```diff
@@ -0,0 +1,83 @@
...
+      :data-testid="dataTestid"
+      :value="value"
+      :rows="rows"
+      :class="{
+        'focus:outline-red-600 outline-red-600': hasError,
+        'dark:outline-slate-600 dark:focus:outline-woot-500 outline-slate-200 focus:outline-woot-500':
+          !hasError,
+        'resize-none': !allowResize,
+      }"
+      class="block w-full px-3 py-3 border-none rounded-xl shadow-sm appearance-none outline outline-1 focus:outline focus:outline-2 text-slate-900 dark:text-slate-100 placeholder:text-slate-400 bg-white dark:bg-slate-800"
```
Comment (pranavrajs) [inline app/javascript/v3/components/Form/Textarea.vue:23]:
```suggestion
      class="block w-full p-3 border-none rounded-xl shadow-sm appearance-none outline outline-1 focus:outline focus:outline-2 text-slate-900 dark:text-slate-100 placeholder:text-slate-400 bg-white dark:bg-slate-800"
```

**Review by pranavrajs — CHANGES_REQUESTED** [review]
One minor followup.

**Review by pranavrajs — APPROVED** [review]

**Pull request merged**